compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td

```diff
@@ -322,8 +322,8 @@ def IREEGPU_GPUPadLayoutAttr : AttrDef<IREEGPU_Dialect, "GPUPadLayout"> {
   let parameters = (ins
     // Relevant target properties that will later allow us to decide the
     // serialized pad layout.
-    "uint32_t":$cache_line_bytes,
-    "uint32_t":$cache_sets
+    OptionalParameter<"std::optional<uint32_t>">:$cache_line_bytes,
+    OptionalParameter<"std::optional<uint32_t>">:$cache_sets
```

Comment:
doesn't this create double optionals here? I never remember what the rules are -- does `OptionalParameter` require for the value type to be 'nullable'?

Reply:
> When using OptionalParameter, the default value is set to the C++ default-constructed value for the C++ storage type. For example, Optional<int> will be set to std::nullopt and Attribute will be set to nullptr. The presence of these parameters is tested by comparing them to their “null” values.

It'd set it to `uint32_t()`, and what we want is to make it null. I think there is a difference between zero and null.

https://mlir.llvm.org/docs/DefiningDialects/AttributesAndTypes/#optional-and-default-valued-parameters